examples/src/bin/ech-client.rs

```diff
@@ -201,37 +215,50 @@ struct Args {
     inner_hostname: String,
 }
 
+/// Collect up all `EchConfigListBytes` found in the HTTPS record(s) for a given domain name/port.
+///
+/// The domain name should be the **inner** name used for Encrypted Client Hello (ECH). The
+/// lookup is done using DNS-over-HTTPS to protect that inner name from being disclosed in
+/// plaintext ahead of the TLS handshake that negotiates ECH for the inner name.
+///
+/// Returns an empty vec if no HTTPS records with ECH configs are found.
 // TODO(@cpu): consider upstreaming to hickory-dns
 async fn lookup_ech_configs(
     resolver: &TokioResolver,
     domain: &str,
     port: u16,
-) -> Option<EchConfigListBytes<'static>> {
-    // For non-standard ports, lookup the ECHConfig using port-prefix naming
-    // See: https://datatracker.ietf.org/doc/html/rfc9460#section-9.1
+) -> Vec<EchConfigListBytes<'static>> {
     let qname_to_lookup = match port {
         443 => domain.to_owned(),
         port => format!("_{port}._https.{domain}"),
     };
 
-    resolver
+    let Ok(lookup) = resolver
         .lookup(qname_to_lookup, RecordType::HTTPS)
         .await
-        .ok()?
-        .record_iter()
-        .find_map(|r| match r.data() {
-            RData::HTTPS(svcb) => svcb
-                .svc_params()
+    else {
+        return Vec::new();
+    };
+
+    let mut ech_config_lists: Vec<EchConfigListBytes> = Vec::new();
+    for r in lookup.record_iter() {
+        let RData::HTTPS(svcb) = r.data() else {
+            continue;
+        };
+
+        ech_config_lists.extend(
+            svcb.svc_params()
                 .iter()
-                .find_map(|sp| match sp {
+                .filter_map(|params| match params {
                     (SvcParamKey::EchConfigList, SvcParamValue::EchConfigList(e)) => {
-                        Some(e.clone().0)
+                        Some(e.clone().0.into())
```

Comment:
Here, too, I feel the code would be easier to follow if you skipped the type annotation on `ech_config_lists` definition in favor of calling `EchConfigListBytes::from()` explicitly.

Reply:
Fixed.